### DISCUSSION TITLE
How do i make a collection of a generic model

### DISCUSSION THREAD

**Discussion by janek515** [TypedProject/ts-express-decorators#2154] [Q&A] [answered]
Hi I have an array of a generic type in a model and want to make validation work, but ajv doesn't seem to like it. I couldn't find anything more in docs.
```ts
@Generics("Inner")
export class Outer<Inner> {
    // omitted

    @GenericOf("Inner")
    @CollectionOf("Task<Inner>")
    tasks: Task<Inner>[];
}
```
Gives error:
```
can't resolve reference Task<Inner> from id #
```

**Comment by Romakita** [comment, marked as answer]
Hello @janek515 

look the pagination example on the page (Pagination.ts)
https://tsed.io/docs/model.html#pagination

See you
